From kyle@athena.road.mcmartin.ca Tue Jun 27 09:11:48 2006
Date: Sun, 25 Jun 2006 18:45:29 -0400
From: Kyle McMartin <kyle@mcmartin.ca>
To: gregkh@suse.de
Subject: USB: Kill compiler warning in quirk_usb_handoff_ohci
Message-ID: <20060625224529.GB2837@athena.road.mcmartin.ca>
Content-Disposition: inline

From: Kyle McMartin <kyle@parisc-linux.org>

Move variables only used on !__hppa__ into that #ifndef section. This
cleans up a compiler warning on parisc. Problem pointed out by
Joel Soete.

Signed-off-by: Kyle McMartin <kyle@parisc-linux.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/pci-quirks.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/pci-quirks.c
+++ gregkh-2.6/drivers/usb/host/pci-quirks.c
@@ -168,8 +168,6 @@ static int __devinit mmio_resource_enabl
 static void __devinit quirk_usb_handoff_ohci(struct pci_dev *pdev)
 {
 	void __iomem *base;
-	int wait_time;
-	u32 control;
 
 	if (!mmio_resource_enabled(pdev, 0))
 		return;
@@ -180,9 +178,10 @@ static void __devinit quirk_usb_handoff_
 
 /* On PA-RISC, PDC can leave IR set incorrectly; ignore it there. */
 #ifndef __hppa__
-	control = readl(base + OHCI_CONTROL);
+{
+	u32 control = readl(base + OHCI_CONTROL);
 	if (control & OHCI_CTRL_IR) {
-		wait_time = 500; /* arbitrary; 5 seconds */
+		int wait_time = 500; /* arbitrary; 5 seconds */
 		writel(OHCI_INTR_OC, base + OHCI_INTRENABLE);
 		writel(OHCI_OCR, base + OHCI_CMDSTATUS);
 		while (wait_time > 0 &&
@@ -199,6 +198,7 @@ static void __devinit quirk_usb_handoff_
 		/* reset controller, preserving RWC */
 		writel(control & OHCI_CTRL_RWC, base + OHCI_CONTROL);
 	}
+}
 #endif
 
 	/*
